core/util.go

```diff
@@ -295,17 +295,17 @@ func VerifyCSR(csr *x509.CertificateRequest) error {
 // SerialToString converts a certificate serial number (big.Int) to a String
 // consistently.
 func SerialToString(serial *big.Int) string {
-	return fmt.Sprintf("%032x", serial)
+	return fmt.Sprintf("%036x", serial)
```

Comment:
Above in `CA` you've used `X` instead of `x`, either this or that should be switched so the format is standardized.